libs/map/gps_track.cpp

```diff
@@ -126,18 +96,24 @@ void GpsTrack::ScheduleTask()
 {
   lock_guard<mutex> lg(m_threadGuard);
 
+  /// @todo Replace with !m_thread.joinable() ?
   if (m_thread.get_id() == std::thread::id())
   {
     m_thread = threads::SimpleThread([this]()
     {
       unique_lock<mutex> ul(m_threadGuard);
       while (true)
       {
-        m_cv.wait(ul, [this]() -> bool { return m_threadExit || m_threadWakeup; });
+        m_cv.wait(ul, [this]() { return m_threadExit || m_threadWakeup; });
+
+        if (m_threadWakeup)
+        {
+          m_threadWakeup = false;
+          ProcessPoints();
```

Comment:
In theory, ProcessPoints may fire the callback in the GpsTrack destructor, when GUI and callback may not be available anymore, so calling it _after_ m_threadExit check is safer.

In practice, GpsTrack lives in the static GpsTracker that is never destroyed…

Reply:
Finish current points first, then exit.